### PR TITLE
[parity-crypto] Add blake2 compression function, aka "F"

### PR DESCRIPTION
**Pull request by dvdplm**
Copies the impl from https://github.com/paritytech/parity-ethereum/pull/11009 and hooks it up in `parity-crypto`.

Supersedes https://github.com/paritytech/parity-ethereum/pull/11009 (probably? cc @seunlanlege )

Related to https://github.com/paritytech/parity-ethereum/issues/10994

### CODE REVIEW

**Comment by DemiMarie-parity** [comment]
This implementation is not efficient ― there are far faster SIMD-optimized implementations.

**Comment by dvdplm** [comment]
> This implementation is not efficient ― there are far faster SIMD-optimized implementations.

You are absolutely correct, it is not efficient. This impl will be replaced by an efficient version asap, but we need something that is 1. correct and 2. easy to check against other impls so we can make progress. We have a pretty tight deadline on the Istanbul work.

**Comment by niklasad1** [comment]
> You are absolutely correct, it is not efficient. This impl will be replaced by an efficient version asap, but we need something that is 1. correct and 2. easy to check against other impls so we can make progress. We have a pretty tight deadline on the Istanbul work.

Please open an issue to this, so we keep track of it

File: parity-crypto/src/blake2_f.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
+/// the precomputed values for BLAKE2b
```
Comment (ordian) [inline parity-crypto/src/blake2_f.rs]: License header is missing

**Comment by dvdplm** [comment]
Closing in favour of https://github.com/paritytech/parity-ethereum/pull/11017 where the `Blake2_f` stuff will live in its own crate in `util/eip-152`.

**Comment by dvdplm** [comment]
@niklasad1 After https://github.com/paritytech/parity-ethereum/pull/11017 lands (before Friday 6) we'll see where we're at and add issue(s) as needed for sure.

**Pull request closed without merging**